kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -208,36 +214,69 @@ private void initialize() {
-        otherNameText.textProperty().bind(otTextProperty);
+
+        // This will listen to the pattern descriptions event. Adding an FQN, Adding other name.
         patternDescriptionEventSubscriber = evt -> {
+            // This if is invoked when the data is coming from FQN name screen.
             if (evt.getEventType() == PatternDescriptionEvent.PATTERN_ADD_FQN) {
-                patternViewModel.setFullyQualifiedName(evt.getDescrName());
-                addFqnMenuItem.setVisible(fqnTextProperty.isEmpty().get());
+                DescrName descrName = evt.getDescrName();
+                patternViewModel.setPropertyValue(FQN_DESCRIPTION_NAME_TEXT, descrName.getNameText());
+                patternViewModel.setPropertyValue(FQN_DESCRIPTION_NAME, descrName);
             }
+
+            // This if is invoked when the data is coming from Other name name screen.
             if (evt.getEventType() == PatternDescriptionEvent.PATTERN_ADD_OTHER_NAME) {
-                patternViewModel.setOtherNameText(evt.getDescrName());
-            }
-            String dateAddedStr = LocalDate.now().format(DateTimeFormatter.ofPattern("MMM d, yyyy")).toString();
-            if (patternViewModel.getProperty(FQN_DESCRIPTION_NAME_TEXT) != null
-                && !((String)patternViewModel.getPropertyValue(FQN_DESCRIPTION_NAME_TEXT)).isEmpty()) {
-                fqnDate.setText("Date Added: " + dateAddedStr);
-                fqnDate.getStyleClass().add("text-noto-sans-normal-grey-eight");
-            } else {
-                fqnDate.setText("");
-            }
-            if (patternViewModel.getProperty(OTHER_NAME_DESCRIPTION_NAME_TEXT) != null
-                && !((String)patternViewModel.getPropertyValue(OTHER_NAME_DESCRIPTION_NAME_TEXT)).isEmpty()) {
-                otherNameDate.setText("Date Added: " + dateAddedStr);
-                otherNameDate.getStyleClass().add("text-noto-sans-normal-grey-eight");
-            } else {
-                otherNameDate.setText("");
+                ObservableList<DescrName> descrNameObservableList = patternViewModel.getObservableList(OTHER_NAMES);
+                descrNameObservableList.add(evt.getDescrName());
             }
         };
+
         EvtBusFactory.getDefaultEvtBus().subscribe(patternViewModel.getPropertyValue(PATTERN_TOPIC), PatternDescriptionEvent.class, patternDescriptionEventSubscriber);
 
+        ObjectProperty<DescrName> objectProperty = patternViewModel.getProperty(FQN_DESCRIPTION_NAME);
+        objectProperty.addListener(new ChangeListener() {
+            @Override
+            public void changed(ObservableValue observableValue, Object o, Object t1) {
+                DescrName descrName = (DescrName) t1;
+                fqnDescriptionSemanticText.setText(" (%s)".formatted(generateDescriptionSemantics(descrName)));
+                fqnDescriptionSemanticText.getStyleClass().add("descr-semantic");
```

Comment:
Since there is only one `fqnDescriptionSemanticText` the style class should have been already added either in SceneBuilder(FXML) or in the `initialize()` method.

Reply:
Yes, removed the styleclass.